Binary_Search/Binary_Search.dart

```diff
@@ -0,0 +1,175 @@
+import 'dart:io';
+
+void binarySearch(List<int> list, int x){
+    int size = list.length;
+    int low = 0;
+    int high = size - 1;
+    while(low <= high){
+        int mid = ((low + high) / 2).floor();
+        if(list[mid] == x){
+            print('Found $x in the List at Index $mid!');
+            return;
+        }
+        if(list[mid] > x){
+            high = mid - 1;
+            continue;
+        }
+        low = mid + 1;
+    }
+    print('Did not $x in the List!');
+    return;
+}
+
+main(){
+    List<int> list = new List();
+    int task = 0;
+    print('Binary Search Program');
+    while(task != 4){
+        print('0 - Display List\n1 - Insert element in List \n2 - Delete element in list \n3 - Search element in List\n4 - Quit\nSelect task:-');
+        try{
+            task = int.parse(stdin.readLineSync());
+        } on Exception{
+            print('Please enter valid Integer value!');
+            continue;
+        }
+        switch(task){
+            case 0:
+                if(list.length == 0){
+                    print('Empty List!');
+                } 
+                else{
+                    print('List:-');
+                    list.forEach((val){
+                        print(val);
+                    });
+                }
+                break;
+            case 1:
+                int val = null;
+                print('Enter value that needs to be inserted in List:- ');
+                try{
+                    val = int.parse(stdin.readLineSync());
+                } on Exception{
+                    print('Please enter valid Integer value!');
+                    continue;
+                }
+                if(val == null){
+                    print('Failed to insert element in List!');
+                    continue;
+                }
+                list.add(val);
+                list.sort();
+                print('Successfully inserted $val in List!');
+                break;
+
+            case 2:
+                int val = null;
```

Comment:
Hi, Delete functionality is not needed in this program. It is for doing binary search alone. Please remove the switch case for delete.

Reply:
Deleted